packages/trader/src/Modules/Contract/Components/InfoBox/__tests__/info-box-longcode.spec.tsx

```diff
@@ -1,56 +1,97 @@
 import React from 'react';
 import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { isMobile } from '@deriv/shared';
+import { mockStore, StoreProvider } from '@deriv/stores';
 import InfoBoxLongcode from '../info-box-longcode';
 
 const test_longcode_short = 'test longcode';
 const test_longcode_long =
     'test longcode test longcode test longcode test longcode test longcode test longcode test longcode test longcode test longcode test longcode test longcode test longcode';
 const test_longcode_mobile =
     'test longcode test longcode test longcode test longcode test longcode test longcode test longcode test longcode';
+const view_more_text = /View more/i;
+const view_less_text = /View less/i;
 const mocked_props = {
     contract_info: { longcode: test_longcode_short, contract_type: 'test' },
 };
 jest.mock('@deriv/components', () => ({
     ...jest.requireActual('@deriv/components'),
     Icon: jest.fn(() => 'MockedIcon'),
 }));
-jest.mock('@deriv/shared', () => ({
-    ...jest.requireActual('@deriv/shared'),
-    isMobile: jest.fn(() => false),
-}));
 
 describe('InfoBoxLongcode', () => {
+    const MockInfoBoxLongcode = (mocked_props: React.ComponentProps<typeof InfoBoxLongcode>) => {
+        const mocked_store = mockStore({
+            ui: {
+                is_mobile: false,
+            },
+        });
+        return (
+            <StoreProvider store={mocked_store}>
```

Comment:
You can pass `mocked_store` as an argument. In this case there will be no need in writing `<StoreProvider>` one more time on lines 67 and 84

Reply:
I'm not getting you actually, I already pass it as an argument, and the reason I don't reuse it on 67 and 84 is because I modify params there 😄

Reply:
```
const MockInfoBoxLongcode = (mocked_store:TCoreStore, mocked_props: React.ComponentProps<typeof 
    InfoBoxLongcode>) => {
        return (
            <StoreProvider store={mocked_store}>
                            <InfoBoxLongcode {...mocked_props} />
            </StoreProvider>
        );
  }
```
...
```
 it('should render specific text "View more" if longcode is more then 47 symbols for mobile', () => {
        const mocked_store = mockStore({
            ui: {
                is_mobile: true,
            },
        });
        mocked_props.contract_info.longcode = test_longcode_mobile;
       render(MockInfoBoxLongcode(mockStore(mocked_root_store), mocked_props));

        expect(screen.getByText(view_more_text)).toBeInTheDocument();
     }
```
    ...
   For default cases, where `is_mobile`: `false` you can create a constant and pass it into `MockInfoBoxLongcode`

Reply:
Just suggestion for future